config/configrotate/configrotate.go

```diff
@@ -0,0 +1,54 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package configrotate // import "go.opentelemetry.io/collector/config/configrotate"
+
+import (
+	"io"
+	"os"
+
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+type Config struct {
+	// Enabled controls whether or not rotate logs
+	Enabled bool `mapstructure:"enable"`
+
+	// MaxMegabytes is the maximum size in megabytes of the file before it gets
+	// rotated. It defaults to 100 megabytes.
+	MaxMegabytes int `mapstructure:"max_megabytes"`
+
+	// MaxDays is the maximum number of days to retain old log files based on the
+	// timestamp encoded in their filename.  Note that a day is defined as 24
+	// hours and may not exactly correspond to calendar days due to daylight
+	// savings, leap seconds, etc. The default is not to remove old log files
+	// based on age.
+	MaxDays int `mapstructure:"max_days"`
```

Comment:
So what is the default, 0?

Reply:
Yes.

https://github.com/open-telemetry/opentelemetry-collector/blob/86f33b34b85681d316cb32fe0df25d8321cb6476/otelcol/unmarshaler.go#L52-L58

As there is no ServiceFactory. Default settings of service are in the unmarshaller.